app/assets/javascripts/lib/bip.js

```diff
@@ -1,780 +0,0 @@
-      }
-      output.append(input_elt);
-      if(this.okButton) {
-        output.append(
-          jQuery(document.createElement('input'))
-          .attr('type', 'submit')
-          .attr('class', this.okButtonClass)
-          .attr('value', this.okButton)
-        )
-      }
-      if(this.cancelButton) {
-        output.append(
-          jQuery(document.createElement('input'))
-          .attr('type', 'button')
-          .attr('class', this.cancelButtonClass)
-          .attr('value', this.cancelButton)
-        )
-      }
-
-      this.element.html(output);
-      this.setHtmlAttributes();
-      // START METAMAPS CODE
-      //this.element.find("input[type='text']")[0].select();
-      this.element.find("input[type='text']")[0].focus();
-      // END METAMAPS CODE
-      this.element.find("form").bind('submit', {editor: this}, BestInPlaceEditor.forms.input.submitHandler);
-      if (this.cancelButton) {
-        this.element.find("input[type='button']").bind('click', {editor: this}, BestInPlaceEditor.forms.input.cancelButtonHandler);
-      }
-      this.element.find("input[type='text']").bind('blur', {editor: this}, BestInPlaceEditor.forms.input.inputBlurHandler);
-      // START METAMAPS CODE
-      this.element.find("input[type='text']").bind('keydown', {editor: this}, BestInPlaceEditor.forms.input.keydownHandler);  
-      // END METAMAPS CODE
-      this.element.find("input[type='text']").bind('keyup', {editor: this}, BestInPlaceEditor.forms.input.keyupHandler);
-      this.blurTimer = null;
-      this.userClicked = false;
-    },
-
-    getValue : function() {
-      return this.sanitizeValue(this.element.find("input").val());
-    },
-
-    // When buttons are present, use a timer on the blur event to give precedence to clicks
-    inputBlurHandler : function(event) {
-      if (event.data.editor.okButton) {
-        event.data.editor.blurTimer = setTimeout(function () {
-          if (!event.data.editor.userClicked) {
-            event.data.editor.abort();
-          }
-        }, 500);
-      } else {
-        if (event.data.editor.cancelButton) {
-          event.data.editor.blurTimer = setTimeout(function () {
-            if (!event.data.editor.userClicked) {
-              event.data.editor.update();
-            }
-          }, 500);
-        } else {
-          event.data.editor.update();
-        }
-      }
-    },
-
-    submitHandler : function(event) {
-      event.data.editor.userClicked = true;
-      clearTimeout(event.data.editor.blurTimer);
-      event.data.editor.update();
-    },
-
-    cancelButtonHandler : function(event) {
-      event.data.editor.userClicked = true;
-      clearTimeout(event.data.editor.blurTimer);
-      event.data.editor.abort();
-      event.stopPropagation(); // Without this, click isn't handled
-    },
-
-    keyupHandler : function(event) {
-      if (event.keyCode == 27) {
-        event.data.editor.abort();
-      }
-      // START METAMAPS CODE
```

Comment:
this is the enter handler. i'm gonna try to see if i can do it with an external event handler, but i might fail